File: Java/labs/lab21_Strategy/task_2_3/src/File.java
public class File {

    final private String fileName;

    enum EncryptionType {
        AES,
        DES,
        RSA,
        ECC
    }

    public File(String fileName) {
        this.fileName = fileName;
    }

    public void encrypt(EncryptionType encryptionType) {
        switch (encryptionType) {

            case AES:
                System.out.println("Applying AES encryption to " + fileName);
                break;
            case DES:
                System.out.println("Applying DES encryption to " + fileName);
                break;
            case RSA:
                System.out.println("Applying RSA encryption to " + fileName);
                break;
            case ECC:
                System.out.println("Applying ECC encryption to " + fileName);
                break;
        }


    }

}

File: Cpp/lab21_Strategy/task_2_3/src/file.cpp
#include <iostream>

#include "file.hpp"


File::File(std::string const& fileName)
: fileName(fileName) {}

void File::encrypt(File::EncryptionType encryptionType)
{
  switch (encryptionType)
  {
    case File::EncryptionType::AES:
    {
      std::cout << "Applying AES encryption to " << fileName << std::endl;
      break;
    }
    case File::EncryptionType::DES:
    {
      std::cout << "Applying DES encryption to " << fileName << std::endl;
      break;
    }
    case File::EncryptionType::RSA:
    {
      std::cout << "Applying RSA encryption to " << fileName << std::endl;
      break;
    }
    case File::EncryptionType::ECC:
    {
      std::cout << "Applying ECC encryption to " << fileName << std::endl;
      break;
    }
  }
}
